src/ast/irbuilderbpf.cpp

```diff
@@ -906,6 +906,19 @@ CallInst *IRBuilderBPF::CreateGetUidGid()
   return createCall(getuidgid_func, {}, "get_uid_gid");
 }
 
+CallInst *IRBuilderBPF::CreateGetNumaId()
+{
+  // u32 bpf_get_numa_node_id(void)
+  // Return: NUMA Node ID
+  FunctionType *getnumaid_func_type = FunctionType::get(getInt32Ty(), false);
+  PointerType *getnumaid_func_ptr_type = PointerType::get(getnumaid_func_type, 0);
+  Constant *getnumaid_func = ConstantExpr::getCast(
+      Instruction::IntToPtr,
+      getInt64(libbpf::BPF_FUNC_get_numa_node_id),
+      getnumaid_func_ptr_type);
+  return createCall(getnumaid_func, {}, "get_numa_id");
```

Comment:
Missing upcast from `i32` to `i64` here. Somthing like:
```suggestion
  auto *call = createCall(getnumaid_func, {}, "get_numa_id");
  return CreateZExt(call, getInt64Ty());
```

Reply:
Marking as unresolved, still missing the upcast as suggested.

Reply:
still applies right?